astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -631,11 +631,30 @@ def _create_dbs(client: ClickHouseClient) -> Iterator[Awaitable[None]]:
             # we need to re-enable these custom global settings when creating the table again.
             # We can enable these settings unconditionally because they are harmless
             # for tables not needing them.
+            # Upstream has introduced a similar list in the replica recovery context:
+            # https://github.com/ClickHouse/ClickHouse/commit/48ed54e822f6ed6f6bdd67db3df7a4a58e550bbb
+            # the two should be kept in sync until system.tables captures the query context.
+            b"SET allow_experimental_inverted_index=true",
+            b"SET allow_experimental_codecs=true",
+            b"SET allow_experimental_live_view=true",
+            b"SET allow_experimental_window_view=true",
+            b"SET allow_experimental_funnel_functions=true",
+            b"SET allow_experimental_nlp_functions=true",
+            b"SET allow_experimental_hash_functions=true",
+            b"SET allow_experimental_object_type=true",
             b"SET allow_experimental_annoy_index=true",
+            b"SET allow_experimental_usearch_index=true",
+            b"SET allow_experimental_bigint_types=true",
+            b"SET allow_experimental_window_functions=true",
             b"SET allow_experimental_geo_types=true",
-            b"SET allow_experimental_object_type=true",
-            b"SET allow_suspicious_codecs=true",
+            b"SET allow_experimental_map_type=true",
             b"SET allow_suspicious_low_cardinality_types=true",
+            b"SET allow_suspicious_fixed_string_types=true",
+            b"SET allow_suspicious_indices=true",
+            b"SET allow_suspicious_codecs=true",
+            b"SET allow_hyperscan=true",
+            b"SET allow_simdjson=true",
+            b"SET allow_deprecated_syntax_for_merge_tree=true",
```

Comment:
Good work. The only thing I am concerned about is compatibility.
Previously the list was conservative and now it is quite wide.
It is possible that some very new option is not available on a running clickhouse, and will cause error like this
```
clickhouse01 :) set test=1

Received exception from server (version 24.8.4):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting 'test'. (UNKNOWN_SETTING)
```
Which is extremely unpleasant during restoration.
I think it makes sense to just ignore those errors or check if the setting is available using `system.settings` table.

Reply:
> unless we manually call /restore on a running node

and that's actually also an edge case for a future update of that list, since the old nodes don't have the extended list and the new nodes already have 23.8 LTS

Reply:
You are right, but I think it is easier to just add the safety guard and forget about possible problems:
1. may be someone is using this open source project)
2. it is actually better to add all the settings from the latest clickhouse, rather then currently used lts
3. situation may change in the future it is better to just be prepared
I really do not see a reason to not protect this place.